tests/python/pants_test/util/test_objects.py

```diff
@@ -709,53 +709,47 @@ def test_enum_class_creation_errors(self):
       "When converting all_values ([1, 2, 3, 1]) to a set, at least one duplicate "
       "was detected. The unique elements of all_values were: [1, 2, 3].")
     with self.assertRaisesRegexp(ValueError, expected_rx):
-      class DuplicateAllowedValues(enum('x', [1, 2, 3, 1])): pass
+      class DuplicateAllowedValues(enum([1, 2, 3, 1])): pass
 
   def test_enum_instance_creation(self):
-    self.assertEqual(1, SomeEnum.create().x)
-    self.assertEqual(2, SomeEnum.create(2).x)
-    self.assertEqual(1, SomeEnum(1).x)
+    self.assertEqual(2, SomeEnum(2).x)
 
-  def test_enum_instance_creation_errors(self):
     expected_rx = re.escape(
       "Value 3 for 'x' must be one of: [1, 2].")
-    with self.assertRaisesRegexp(EnumVariantSelectionError, expected_rx):
-      SomeEnum.create(3)
     with self.assertRaisesRegexp(EnumVariantSelectionError, expected_rx):
       SomeEnum(3)
 
     # Specifying the value by keyword argument is not allowed.
     with self.assertRaisesRegexp(TypeError, re.escape("__new__() got an unexpected keyword argument 'x'")):
       SomeEnum(x=3)
 
-    # Test that None is not used as the default unless none_is_default=True.
-    with self.assertRaisesRegexp(EnumVariantSelectionError, re.escape(
-        "Value None for 'x' must be one of: [1, 2]."
-    )):
-      SomeEnum.create(None)
-    self.assertEqual(1, SomeEnum.create(None, none_is_default=True).x)
-
-    expected_rx_falsy_value = re.escape(
-      "Value {}'' for 'x' must be one of: [1, 2]."
-      .format('u' if PY2 else ''))
-    with self.assertRaisesRegexp(EnumVariantSelectionError, expected_rx_falsy_value):
-      SomeEnum('')
+  def test_enum_generated_attrs(self):
+    class HasAttrs(enum(['a', 'b'])): pass
+    self.assertEqual(HasAttrs.a, HasAttrs('a'))
+    self.assertEqual(type(HasAttrs.a), HasAttrs)
+    self.assertEqual(HasAttrs.b, HasAttrs('b'))
 
-  def test_enum_comparison_fails(self):
+  def test_enum_comparison(self):
     enum_instance = SomeEnum(1)
-    rx_str = re.escape("enum equality is defined to be an error")
-    with self.assertRaisesRegexp(TypeCheckError, rx_str):
-      enum_instance == enum_instance
-    with self.assertRaisesRegexp(TypeCheckError, rx_str):
-      enum_instance != enum_instance
-    # Test that comparison also fails against another type.
+    another_enum_instance = SomeEnum(2)
+    self.assertEqual(enum_instance, enum_instance)
+    self.assertNotEqual(enum_instance, another_enum_instance)
+
+    # Test that comparison fails against another type.
+    rx_str = re.escape(
+      "when comparing against 1 with type 'int': "
+      "enum equality is only defined for instances of the same enum class!")
     with self.assertRaisesRegexp(TypeCheckError, rx_str):
```

Comment:
Aside for another time: I've seen this pattern enough that it's probably adding to our `TestBase` a `self.assertRaisesWithMessageContaining(exception_type, substring)`, to avoid needing to do the `re.escape` dance in tests (and make it clear to the reader that we're not actually doing regex comparisons)

Reply:
Yes, I've wanted to rewrite all of `test_objects.py` many times and never did it because I always got distracted by more changes to `objects.py` I wanted to make. This can be broken out into an issue.

Reply:
Made #7298!